app/models/user.rb

```diff
@@ -200,123 +229,72 @@ def can_make_challenges
   end
 
-  def has_clearance?(permission_level)
+  def clearance?(permission_level)
     # for each group (devs, mods, admins) there is a bit in the permission_level
     # indicating whether that group has clearance.
     # a user has clearance if they are a member of any of the groups which has clearance.
     # if all bits are set, then everyone has clearance (including users not in any group);
     # if no bits are set, then no-one has clearance.
     everyone_clearance = permission_level == ~0
-    admin_clearance    = permission_level.anybits?(1 << 0)
-    mod_clearance      = permission_level.anybits?(1 << 1)
-    dev_clearance      = permission_level.anybits?(1 << 2)
+    admin_clearance = permission_level.anybits?(1 << 0)
+    mod_clearance = permission_level.anybits?(1 << 1)
+    dev_clearance = permission_level.anybits?(1 << 2)
 
     everyone_clearance ||
       (is_developer && dev_clearance) ||
       (is_moderator && mod_clearance) ||
-      (is_admin     && admin_clearance)
-  end
-
-  def submission_limit
-    max_dad_level = highest_level
-    return 1 if max_dad_level.zero?
-
-    return max_dad_level if max_dad_level < UNLIMITED_SUBMISSIONS_REQUIREMENT
-
-    -1
+      (is_admin && admin_clearance)
   end
 
   def profile_picture
-    profile_picture = avatar.thumb.url unless avatar.nil?
-
-    if profile_picture.blank?
-      return 'https://s3.us-east-2.amazonaws.com/do-art-daily-public/Default+User+Thumb.png'
-    end
-
-    profile_picture
+    avatar.thumb.url || 'https://s3.us-east-2.amazonaws.com/do-art-daily-public/Default+User+Thumb.png'
   end
 
+  ### Moderation ###
   def approve(reason, moderator)
-    ModeratorLog.create(user_id: moderator.id,
-                        target: self,
-                        action: "#{moderator.username} has approved #{username}.",
-                        reason: reason)
-    update_attribute(:approved, true)
+    create_mod_log(moderator, "#{moderator.username} has approved #{username}.", reason)
+    update_retain_password(approved: true)
+  end
+
+  def latest_ban
+    site_ban.find_by("'#{Time.now.utc}' < expiration")
   end
 
   def lift_ban(reason, moderator)
     # Do not lift bans on users marked for death.
-    unless marked_for_death
-      site_ban = SiteBan.find_by("'#{Time.now.utc}' < expiration AND user_id = #{id}")
-      unless site_ban.nil?
-        ModeratorLog.create(user_id: moderator.id,
-                            target: self,
-                            action: "#{moderator.username} has lifted #{username}'s ban (original expiry: #{ApplicationController.helpers.date_string_short(site_ban.expiration)}).",
-                            reason: reason)
-        site_ban.destroy
-      end
-    end
+    return if marked_for_death || latest_ban.nil?
+
+    action = "#{moderator.username} has lifted #{username}'s ban " \
+              "(original expiry: #{latest_ban.expiration})."
+    create_mod_log(moderator, action, reason)
```

Comment:
Similarly, the create_mod_log method here omits the target.

Reply:
Same as above